setup/src/installer.ts

```diff
@@ -220,7 +226,37 @@ async function ghcup(tool: Tool, version: string, os: OS): Promise<void> {
   if (returnCode === 0) await exec(bin, ['set', tool, version]);
 }
 
-function getChocoPath(tool: Tool, version: string): string {
+async function getChocoPath(
+  tool: Tool,
+  version: string
+): Promise<string | null> {
+  if (!process.env.ChocolateyInstall) {
+    return null;
+  }
+
+  if (tool === 'cabal') {
+    const chocoToolPath = join(
+      `${process.env.ChocolateyInstall}`,
+      'lib',
+      `${tool}.${version}`
+    );
+
+    const globber = await glob.create(`${chocoToolPath}/tools/*/cabal.exe`);
```

Comment:
If we already need to use a globbing function here, we can probably collapse all of the complexity in the function to something like

```typescript
  const chocoToolPath = join(
    `${process.env.ChocolateyInstall}`,
    'lib',
    `${tool}.${version}`,
    'tools',
    '*',
    tool === 'ghc' ? 'bin' : '',
    tool
  );

  const globber = await glob(
    [chocoToolPath, `${chocoToolPath}.exe`].join('\n')
  );
```

but we'd need to be careful to grab the correct version; since we still have the tool.version in the path, I think we're fine there.

Reply:
Simplified the function.